src/knightos/display.c

```diff
@@ -330,6 +330,50 @@ void draw_short(SCREEN *screen, unsigned char x, unsigned char y, unsigned short
 	screen; x; y; value;
 }
 
+void draw_signed(SCREEN* screen, unsigned char x, unsigned char y, signed short value) {
+    if(value < 0){
+        draw_char(screen, x, y, '-');
+        x += 4;
+        value = -value;
+    }
+    draw_short(screen, x, y, value);
+    return;
+}
+
+void draw_float(SCREEN* screen, unsigned char x, unsigned char y, float value){
+    /* Implementation is weird and slow because of non-working snprintf. Indeed, %f format strings are not yet implemented */
+    unsigned int integer_part;
+    unsigned int frac_part;
+    unsigned int integer_part_len;
+    bool sgn = (value < 0);
+    
+    
+    value = sgn ? -value : value;
+    
+    integer_part = (unsigned int)value;
+    frac_part = ((value - (float)integer_part)*10000);
+    
+#define __log10u(x) ((x >= 10000000u) ? 7 : (x >= 1000000u) ? 6 : \
```

Comment:
As this is only used once, there's no real point in it being a macro

Reply:
You have a point. I actually did that one for my infix calculator project for KnightOS (because it seriously needs it).
This function is so useful (and actually so fast... maybe coding it in asm?) that it may be included as a non standard libc extension.
Just my two cents.

Reply:
If you want to add that as a function, feel free to do so :)